### ISSUE TITLE
CMS Plugin for Bluzelle Web Hosting Services on WordPress, Wix, etc

### ISSUE THREAD

**Issue by njmurarka** [bluzelle/Gitcoin10Hack#9]
### Prize Title

CMS Plugin for Bluzelle Web Hosting Services on WordPress, Wix, etc

### Prize Bounty

$2,500 USD worth of ERC20 BLZ for each CMS plugin submission, at time of payout

### Challenge Description

Bluzelle has a web-serving endpoint has has been built as a working POC. It allows an entire website to be hosted upon the database. This is done by allowing each file in the website to be stored as a key and value pair. Files can be referenced relative to the root path of the website, just like any standard web server.

A key value of this service is high availability and censorship-resistance -- a website hosted this way on Bluzelle cannot be taken down by anyone. To do so would require an attack on the network that would have to breach the very core of our ProofOfStake blockchain consensus algorithm. 

Some brief documentation and helper code can be found at:

https://github.com/bluzelle/curium/tree/stargate/sdk/site-demo

Please refer to our site-demo link above, to find an example of an uploader script (file-uploader.ts) that uploads everything in the site folder, to our testnet. 

Once uploaded, the entire website can be accessed with a URL as follows (a real example):

https://client.sentry.testnet.private.bluzelle.com:1317/crud/raw/my-sitex/index.html

We are looking for plugins to be built for leading CMS systems, that, once plugged in, allow someone to build a website via the CMS and then publish it to Bluzelle. So, for example, a plugin for WordPress. Note that our Bluzelle JS libraries are the only ones currently available, so it only be practical to build plugins in JS. 

The above references can be very useful to determine the right commands to build a plugin successfully. 

The service itself always hosts sites on port 1317, and has the format above, for now. In the future, we will try to improve the look of this URL to be more human friendly and possibly even allow CNAME records, etc.

The plugin itself should be simple and straightforward. You install it and configure it with your Bluzelle account settings, UUID, etc, and then can push and even pull content from your website hosted on Bluzelle. 

End result is you can host an entire website on our infrastructure, free from the risk of takedown by any 3rd parties (contingent on a proper decentralized design). This is a particularly powerful API upon which free-speech and other related platforms can be built. While we cannot vouch for something like “Parler”, an app such as this one could be built on our platform and not face the risk of takedown by anybody.

In the event there are multiple submissions per CMS, we reserve the right to choose the best one.

### Resources

Our Discord: 

https://discord.gg/yqBmzPxRZK

Website:

https://www.bluzelle.com

Our JS library:

https://www.npmjs.com/package/@bluzelle/sdk-js

To install our libraries:

install @bluzelle/sdk-js with “yarn” or “npm”

### Submission Requirements

The submission should include sufficiently QA’d documentation on how to deploy the service/product, and how to use the submission as per the requirements of the bounty. 

These should include documentation on the commands to be used to interact with the submission, and how the submission is configured to work properly with BluzelleDB, etc.

A video demo should be included. It would nice to have a voice-over in English where we can fully understand the submission, but this is not a strict requirement. A computer-generated voice over is ok too, if you prefer. 

The demo should also walk through the code and explain all the items that are being provided. The demo should walk through the process of deploying the submission, and how to use it, etc.

It is expected that the documentation is accurate. We will follow your documentation, to properly evaluate the submission. If it is incorrect, we may be unable to fairly evaluate your submission.

Including tests with your submission will greatly improve your chances of winning. We like to run tests that are highly verbose and explicit in terms of what they are doing, so we can gain confidence in the correctness of what you have submitted. If you provide tests and expect us to run them, like everything else, document it well, and ensure that the tests can be run by us -- give us the steps to setup and run the tests.

If the documentation is incomplete or incorrect, there is a possibility that we may not be able to fairly assess the submission, as we will walk through the documentation to validate the project. Due to practical limitations on time and resources, once a project is submitted, we are not able to provide much assistance in correcting a project’s that may not be properly working, nor to inquire to get proper steps, if the documentation that comes with a submission is insufficient. 

Your project will be judged based on what you submit. Please submit something that is complete, well thought out, and tested, from a documentation and product features and code quality standpoint. We will do our best to evaluate, but obviously, the easier you make our life, the better the chances are that you win. 

WE LOVE VERBOSITY AND DOCUMENTATION. There is no such thing as too much information. Explain what you have built, and please ensure it will run CORRECTLY, when we follow your directions literally. Just doing this alone will vastly improve your chances of victory.

### Judging Criteria

Our goal is to, as part of the evaluation process, fully setup, and use your submission, successfully, and without any major hiccups.

Based on the ease of doing this and the quality of your documentation, product, code, and features, we will choose the winner. 

There is no preference to ordering of submissions -- just be sure to submit them on time. Once submitted, we will evaluate and there will not be alot of opportunity for back and forth. Please ensure your submitted documentation and code is complete, enabling us to properly judge it based on its merit.

We will choose the best based on quality. Documentation and properly written code is a large part of the criteria. A project that we cannot deploy ourselves is difficult to give a prize to. We will do our best to contact you, if there is an issue. Practically, we probably won’t have much time to contact you, after submission, to get clarification or to ask you to fix a bug. It ideally should work when we judge it.

Note: While the descriptions given for bounties are quite explicit and even tend to suggest how an actual solution to each problem can be built, you as a developer have the option to architect the solution your own way. We have provided guidance on a solution we see as reasonable, but we are open to considering other solutions. Obviously, we will choose the best overall submission, based on various factors including the elegance of the solution.

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 2500.0 USDC (2500.0 USD @ $1.0/USDC)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $1,933,468.47 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 1 week, 1 day from now.
Please review their action plans below:


**1) [fluffy9](https://gitcoin.co/fluffy9) has started work.**

A Github Action for Bluzelle Web Hosting
**2) [paulohaacke](https://gitcoin.co/paulohaacke) has started work.**

A Plugin for hosting Gatsby (https://www.gatsbyjs.com/) Websites in Bluzelle.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031).

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Work for 2500.0 USDC (2500.00 USD @ $1.0/USDC) has been submitted by__:

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $1,904,593.86 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Work for 2500.0 USDC (2500.00 USD @ $1.0/USDC) has been submitted by__:

 1. [@fluffy9](https://gitcoin.co/fluffy9)



@njmurarka please take a look at the submitted work:
* [PR](https://github.com/Fluffy9/bluzelle) by @fluffy9

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $1,912,201.68 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by translate3d** [comment]
> Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done
> 
> **Work for 2500.0 USDC (2500.00 USD @ $1.0/USDC) has been submitted by**:
> 
> * Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031)
> * Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031).
> * Questions? Checkout [Gitcoin Help](https://gitcoin.co/help) or the [Gitcoin's Discord](https://discord.gg/gitcoin/)
> * $1,904,593.86 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

My name and PR missing there

<img width="1150" alt="Screenshot 2021-07-23 at 14 44 11" src="https://user-images.githubusercontent.com/84582633/126783368-c6f6c93e-b138-45df-954e-2e26218f66cd.png">
. Bad job, @gitcoinbot

**Comment by njmurarka** [comment]
@fluffy9 Please explain how to use your project. No video was provided but is required for the submission.

Thank you.

**Comment by njmurarka** [comment]
> > Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done
> > **Work for 2500.0 USDC (2500.00 USD @ $1.0/USDC) has been submitted by**:
> > 
> > * Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031)
> > * Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/bluzelle/Gitcoin10Hack/9/100026031).
> > * Questions? Checkout [Gitcoin Help](https://gitcoin.co/help) or the [Gitcoin's Discord](https://discord.gg/gitcoin/)
> > * $1,904,593.86 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
> 
> My name and PR missing there
> 
> <img alt="Screenshot 2021-07-23 at 14 44 11" width="1150" src="https://user-images.githubusercontent.com/84582633/126783368-c6f6c93e-b138-45df-954e-2e26218f66cd.png">
> 
> . Bad job, @gitcoinbot

Can you provide a link to your submission? I do not see it.
